src/databricks/labs/ucx/assessment/crawlers.py

```diff
@@ -92,3 +96,72 @@ def spark_version_compatibility(spark_version: str | None) -> str:
     if (10, 0) <= version < (11, 3):
         return "kinda works"
     return "supported"
+
+
+def _check_spark_conf(conf: dict[str, str], source) -> list[str]:
+    failures = []
+    for k in INCOMPATIBLE_SPARK_CONFIG_KEYS:
+        if k in conf:
+            failures.append(f"unsupported config: {k}")
+    for value in conf.values():
+        if "dbfs:/mnt" in value or "/dbfs/mnt" in value:
+            failures.append(f"using DBFS mount in configuration: {value}")
+    # Checking if Azure cluster config is present in spark config
+    if _azure_sp_conf_present_check(conf):
+        failures.append(f"{_AZURE_SP_CONF_FAILURE_MSG} {source}.")
+    return failures
+
+
+def _safe_get_cluster_policy(ws: WorkspaceClient, policy_id: str) -> Policy | None:
```

Comment:
so, logical question - do we want to do diamond shape dependencies or not? e.g. `_safe_get_cluster_policy` - where it should live? in `crawlers.py` or in `clusters.py`? what about `_check_cluster_failures`? 

```mermaid
flowchart TD
    assessment --> crawlers
    crawlers --> clusters
    crawlers --> jobs
    crawlers --> init_scripts
    crawlers --> pipelines
    crawlers --> azure_spns
    clusters --> runtime.py
    jobs --> runtime.py
    init_scripts --> runtime.py
    pipelines --> runtime.py
    azure_spns --> runtime.py
```

or something like this:

```mermaid
flowchart TD
    assessment --> crawlers
    azure_spns -->|mixin| clusters
    crawlers --> clusters
    clusters -->|mixin| jobs
    crawlers --> jobs
    azure_spns -->|mixin| init_scripts
    crawlers --> init_scripts
    jobs -->|mixin| pipelines
    crawlers --> pipelines
    crawlers --> azure_spns
    clusters --> runtime.py
    jobs --> runtime.py
    init_scripts --> runtime.py
    pipelines --> runtime.py
    azure_spns --> runtime.py
```